use requests instead of aiohttp for underlying http client

SebbyLaw opened the pull request:
instead of starting and stopping the current thread's event loop on every http call, just make an http call.

chebbyChefNEQ commented on python/lancedb/remote/client.py:
```diff
@@ -76,38 +76,38 @@ def headers(self) -> Dict[str, str]:
-        async with self.session.get(
-            uri,
+        with self.session.get(
+            self.url + uri,
```
I think we should use `urllib.parse.urljoin` here

chebbyChefNEQ commented on python/lancedb/remote/client.py:
```diff
@@ -76,38 +76,38 @@ def headers(self) -> Dict[str, str]:
             params=params,
             headers=self.headers,
-            timeout=aiohttp.ClientTimeout(total=30),
+            timeout=30,
```
nit: maybe use `(5.0, 30.0)`. 5s for connection timeout and 30s for read timeout

chebbyChefNEQ commented on python/lancedb/remote/client.py:
```diff
@@ -139,31 +139,26 @@ async def post(
-        async with self.session.post(
-            uri,
+        with self.session.post(
+            self.url + uri,
```
ditto: use `urljoin`

chebbyChefNEQ reviewed (commented):
Thank you for doing this!

I think we also need to update `pyproject.toml` to add `requests` as a dependency.

Otherwise looks good

chebbyChefNEQ commented on python/lancedb/remote/client.py:
```diff
@@ -53,15 +53,15 @@ class RestfulLanceDBClient:
-    def session(self) -> aiohttp.ClientSession:
-        url = (
+    def session(self) -> requests.Session:
+        self.url: str = (
             self.host_override
             or f"https://{self.db_name}.{self.region}.api.lancedb.com"
         )
```
let's move this to it's own property

SebbyLaw commented:
> I think we also need to update `pyproject.toml` to add `requests` as a dependency.

requests is already listed as a dependency in the pyproject.toml

I've removed aiohttp from the module dependencies and replaced `requests` with it in the testing dependency extras since it is only used in tests now

chebbyChefNEQ reviewed (approved):
LGTM, barring CI
